src/yunohost/domain.py

```diff
@@ -681,3 +725,296 @@ def _get_DKIM(domain):
                 p=dkim.group("p"),
             ),
         )
+
+
+def _load_domain_settings():
+    """
+    Retrieve entries in domains.yml
+    And fill the holes if any
+    """
+    # Retrieve entries in the YAML
+    old_domains = None
+    if os.path.exists(DOMAIN_SETTINGS_PATH) and os.path.isfile(DOMAIN_SETTINGS_PATH):
+        old_domains = yaml.load(open(DOMAIN_SETTINGS_PATH, "r+"))
+
+    if old_domains is None:
+        old_domains = dict()
+
+    # Create sanitized data
+    new_domains = dict()
+
+    get_domain_list = domain_list()
+
+    # Load main domain
+    maindomain = get_domain_list["main"]
+
+    for domain in get_domain_list["domains"]:
```

Comment:
We should update this to use python's `update` function

Reply:
Done!